app/site/controllers.py

```diff
@@ -128,15 +128,12 @@ def publisher_dashboard(publisher):
 
 @site_blueprint.route("/search", methods=["GET"])
 def search_package():
-    try:
-        q = request.args.get('q')
-        if not q.strip():
-            return render_template("search.html", title='Search', query_term=''), 200
-
+    q = request.args.get('q').strip()
+    if q:
         datapackage_list = DataPackageQuery(query_string=q).get_data()
-        return render_template("search.html",
-                               datapackage_list=datapackage_list,
-                               total_count=len(datapackage_list),
-                               query_term=q), 200
-    except Exception as e:
-        return render_template("search.html", title='Search', query_term=''), 200
+    else:
+        datapackage_list = DataPackageQuery(query_string=q).get_data(50)
```

Comment:
I suggest limiting to 20 results or so. Apologies if i said 50 before - that was just an offhand comment (about having some numbrer)